apps/meteor/ee/app/livechat-enterprise/server/api/rooms.ts

```diff
@@ -1,57 +1,108 @@
-import { Meteor } from 'meteor/meteor';
 import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
-import { isPOSTLivechatRoomPriorityParams } from '@rocket.chat/rest-typings';
+import { isLivechatRoomOnHoldProps, isLivechatRoomResumeOnHoldProps, isPOSTLivechatRoomPriorityParams } from '@rocket.chat/rest-typings';
 import { LivechatRooms } from '@rocket.chat/models';
+import type { IOmnichannelRoom } from '@rocket.chat/core-typings';
+import { isOmnichannelRoom } from '@rocket.chat/core-typings';
+import { OmnichannelEEService } from '@rocket.chat/core-services';
 
 import { API } from '../../../../../app/api/server';
 import { hasPermission } from '../../../../../app/authorization/server';
 import { Subscriptions } from '../../../../../app/models/server';
-import { LivechatEnterprise } from '../lib/LivechatEnterprise';
 import { removePriorityFromRoom, updateRoomPriority } from './lib/priorities';
 
 API.v1.addRoute(
 	'livechat/room.onHold',
-	{ authRequired: true, permissionsRequired: ['on-hold-livechat-room'] },
+	{ authRequired: true, permissionsRequired: ['on-hold-livechat-room'], validateParams: isLivechatRoomOnHoldProps },
 	{
 		async post() {
 			const { roomId } = this.bodyParams;
 			if (!roomId || roomId.trim() === '') {
-				return API.v1.failure('Invalid room Id');
+				throw new Error('error-invalid-room');
 			}
 
-			const room = await LivechatRooms.findOneById(roomId);
-			if (!room || room.t !== 'l') {
-				return API.v1.failure('Invalid room Id');
+			const room = await LivechatRooms.findOneById<Pick<IOmnichannelRoom, '_id' | 't' | 'open' | 'onHold' | 'lastMessage' | 'servedBy'>>(
+				roomId,
+				{
+					projection: { _id: 1, t: 1, open: 1, onHold: 1, lastMessage: 1, servedBy: 1 },
+				},
+			);
+			if (!room || !isOmnichannelRoom(room)) {
+				throw new Error('error-invalid-room');
 			}
 
 			if (room.lastMessage?.token) {
-				return API.v1.failure('You cannot place chat on-hold, when the Contact has sent the last message');
+				throw new Error('error-contact-sent-last-message-so-cannot-place-on-hold');
 			}
 
 			if (room.onHold) {
-				return API.v1.failure('Room is already On-Hold');
+				throw new Error('error-room-is-already-onHold');
 			}
 
 			if (!room.open) {
-				return API.v1.failure('Room cannot be placed on hold after being closed');
+				throw new Error('This_conversation_is_already_closed');
 			}
 
-			const user = Meteor.user();
-			if (!user) {
-				return API.v1.failure('Invalid user');
+			if (!room.servedBy) {
+				throw new Error('error-unserved-rooms-cannot-be-placed-onhold');
 			}
 
-			const subscription = Subscriptions.findOneByRoomIdAndUserId(roomId, user._id, { _id: 1 });
+			const subscription = Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId, { _id: 1 });
```

Comment:
is this part needing to change to `this.userId`?

Reply:
Although its not needed, I still would like to keep this since this is the recommended way to get the logged in user. If you look at the previous code, it had a Meteor dependency which we're trying to avoid as much as possible.